.github/workflows/main.yml

```diff
@@ -6,9 +6,6 @@ on:
       - main
```

Comment:
Let's rename this file to `test.yml` for consistency, and rename the workflow name at the top to `name: test`.

Also, we refer to the workflow in README.rst, that needs to be updated as well:

```diff
diff --git a/README.rst b/README.rst
index f0fe35632..034034a40 100644
--- a/README.rst
+++ b/README.rst
@@ -20,8 +20,8 @@
     :target: https://codecov.io/gh/pytest-dev/pytest
     :alt: Code coverage Status

-.. image:: https://github.com/pytest-dev/pytest/workflows/main/badge.svg
-    :target: https://github.com/pytest-dev/pytest/actions?query=workflow%3Amain
+.. image:: https://github.com/pytest-dev/pytest/workflows/test/badge.svg
+    :target: https://github.com/pytest-dev/pytest/actions?query=workflow%3Atest
```